private/tst/avb_streamhandler/src/IasTestSystemdWatchdog.cpp

```diff
@@ -0,0 +1,230 @@
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  res = altWdManager->destroyWatchdog(wdIf_2);
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+}
+
+TEST_F(IasTestSystemdWatchdog, removeWatchdog)
+{
+  IasWatchdogInterface *wdIf = nullptr;
+  IasWatchdogResult res = IasMediaTransportAvb::IasResult::cOk;
+
+  wdIf = mWdManager->createWatchdog(30000, "Thor");
+  ASSERT_NE(wdIf, nullptr);
+
+  res = mWdManager->removeWatchdog(wdIf);
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  res = mWdManager->removeWatchdog(wdIf);
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cObjectNotFound);
+
+  res = mWdManager->removeWatchdog(nullptr);
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cParameterInvalid);
+}
+
+TEST_F(IasTestSystemdWatchdog, getCurrentRawTime)
+{
+  uint64_t curr_timestamp = 0;
+
+  curr_timestamp = mWdManager->getCurrentRawTime();
+  EXPECT_NE(curr_timestamp, 0);
+}
+
+TEST_F(IasTestSystemdWatchdog, registerWatchdog)
+{
+  IasWatchdogInterface *wdIf = nullptr;
+  IasWatchdogResult res = IasMediaTransportAvb::IasResult::cOk;
+
+  wdIf = mWdManager->createWatchdog();
+  ASSERT_NE(wdIf, nullptr);
+  res = wdIf->registerWatchdog();
+  EXPECT_EQ(res, IasWatchdogResult::cWatchdogNotPreconfigured);
+
+  wdIf->setTimeout(30000);
+  wdIf->setName("Eitri");
+  res = wdIf->registerWatchdog();
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  res = wdIf->unregisterWatchdog();
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  res = wdIf->registerWatchdog(30000, "");
+  EXPECT_EQ(res, IasWatchdogResult::cParameterInvalid);
+
+  res = wdIf->registerWatchdog(0, "Sif");
+  EXPECT_EQ(res, IasWatchdogResult::cParameterInvalid);
+
+  res = wdIf->registerWatchdog(30000, "Sif");
+  EXPECT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  res = wdIf->registerWatchdog(30000, "Sif");
+  ASSERT_EQ(res, IasWatchdogResult::cAlreadyRegistered);
+}
+
+
+TEST_F(IasTestSystemdWatchdog, unregisterWatchdog)
+{
+  IasWatchdogInterface *wdIf = nullptr;
+  IasWatchdogResult res = IasMediaTransportAvb::IasResult::cOk;
+
+  wdIf = mWdManager->createWatchdog(30000, "Loki");
+  ASSERT_NE(wdIf, nullptr);
+
+  res = wdIf->unregisterWatchdog();
+  EXPECT_EQ(res, IasWatchdogResult::cWatchdogUnregistered);
+
+  res = wdIf->registerWatchdog();
+  ASSERT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+}
+
+TEST_F(IasTestSystemdWatchdog, reset)
+{
+  IasWatchdogInterface *wdIf = nullptr;
+  IasWatchdogResult res = IasMediaTransportAvb::IasResult::cOk;
+
+  wdIf = mWdManager->createWatchdog(30000, "Loki");
+  ASSERT_NE(wdIf, nullptr);
+
+  res = wdIf->reset();
+  EXPECT_EQ(res, IasWatchdogResult::cWatchdogUnregistered);
+
+  res = wdIf->registerWatchdog();
+  ASSERT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+  res = wdIf->reset();
+  ASSERT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  sleep(10);
+  res = wdIf->reset();
+  EXPECT_EQ(res, IasWatchdogResult::cTimedOut);
```

Comment:
Maybe I understood it wrong here, but when you created this watchdog with `30000`, it would take 30 seconds to fire, right? Then you sleep for like 10 seconds, and expects it to fail due time out. Will this work? Shouldn't you create the watchdog with a smaller time, like 3 seconds? Then it would have timed out on this test, after 10 seconds waiting. Unless, as I said, I understood things wrong...

Reply:
Ok, seerious and shtuupid bug has been caught. The reset() function does the following:

       _mDueResetTime =  currentTime + (mTimeout * 1000000);_

where mTimeout is what we pass, 30 seconds in this case.

Now, mTimeout has been declared as a uint32_t. For large values like 30 seconds = (30,000 * 1,000,000) nanoseconds, this won't work, obviously. 
I'll include another patch to change mTimeout to a uint64_t. Will reduce the timer here from 30 sec to 5 sec too.

Reply:
Ah, the value of testing =D
Yeah, if we keep time in nanoseconds, `uint64_t` is mandatory. But do we need to keep time in nanoseconds?

Reply:
The comparison against current_time (another uint64_t) won't work then. It's being derived directly from clock_gettime() and the nanosecond component is also being considered there. Also, other parameters such as _dueResetTime_ are also declared as uint64_t. So, changing mTimeout's type looks like the fastest way out, right now.

Reply:
Functionally, I agree, we don't need a nanosecond accuracy here.

Reply:
Fair enough.

Reply:
Done. Thank for the review :)